### PR TITLE
Example showing compile time switch between CPU and GPU use

### PR DESCRIPTION
**Pull request by joscao**
- providing copyable "switch_DMO_CPU_GPU.h" file using alias template  to define a utility::Solver aliasing either DMO::Solver or DMO::SolverCPU
- provide possibility of forcing CPU usage even if GPU is available by additional preprocessor defintion of "_FORCE_CPU_USAGE"
- provide constexpr bool variable utility::use_GPU

### CODE REVIEW

**Comment by DanielZint** [comment]
I like the idea of the example. There just some minor things that confuse me a bit.
* I think variable ```_FORCE_CPU_USAGE``` is not necessary because it does exactly the same as ```_DMO_USE_CUDA```.
* I don't see the advantage of having an individual target for both, GPU and CPU. Both do exactly the same, right?

Furthermore, can you please add some comments to explain your code? As it is an example, I think some more insight in the process of thought would be helpful.

**Comment by joscao** [comment]
- Simplified and documented `switch_DMO_CPU_GPU.h`

Therefore ready to be merged from my side.

File: src/ex03_easy_switch/main.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ * DMO Example 04 - User Metrics on either CPU or GPU chosen at compile time, selecting the one available
```
Comment (daniel-zint) [inline src/ex03_easy_switch/main.cpp:2]: I guess that should be "Example 03"

**Review by daniel-zint — APPROVED** [review]
lgtm

**Pull request closed without merging**